src/Idris/Desugar.idr

```diff
@@ -1023,7 +1027,7 @@ mutual
            syn <- get Syn
            p' <- traverse (desugar AnyExpr ps) field.val.info
            ty' <- bindTypeNames field.fc (usingImpl syn) ps !(desugar AnyExpr ps field.val.boundType)
-           pure (Mk [field.fc, field.rig, n] (MkPiBindData p' ty'))
+           pure (Mk [field.fc, field.rig, AddDef (AddDef n)] (MkPiBindData p' ty'))
```

Comment:
Those `AddDef` are suspicious, are they really necessary?

Reply:
Actually yes, the first one is the default binding and the second is the default documentation